src/components/panels/tools/tools.tsx

```diff
@@ -97,6 +103,21 @@ const ToolsPanel = () => {
 
             <Flex className="tools" overflow="auto">
                 <Accordion width="100%" allowMultiple defaultIndex={[0, 1, 2]}>
+                    <Button
+                        aria-label="select"
+                        leftIcon={selectIcon}
+                        onClick={() => {
+                            if (mode === 'select') {
+                                dispatch(setMode('free'));
```

Comment:
Do we need to set mode to free if the current mode is select? select mode should get cleared when users click on no matter background/canvas or any node/edge. Another thing is do we want to preserve select mode after a drag-to-selct operation? If so, we may set the mode back to free so that users can return to free mode and drag the background/canvas.

Reply:
I think it should be cleared when background/canvas/node/edge is clicked.

Reply:
And cleared to free when a selection finishes.

Reply:
So we can remove this line? As we can assure users can drag background/canvas/node/edge in select mode by clicking on background/canvas/node/edge.

Reply:
Let me have a try.

~~let me see see~~

Reply:
Hmm, I think it would be better to allow users to click Select button to cancel select operation.

Reply:
Okay. We may add some indicators showing the current mode. Maybe darken the select button? Also, make this a one liner like setMode(mode === xxx ? xxx, xxx).

Reply:
In L116, it is different when the mode is set to "select".